front/app/modules/commercial/project_folders/permissions/roles.ts

```diff
@@ -2,14 +2,12 @@ import { isNilOrError } from 'utils/helperUtils';
 import { IUserData, IRole } from 'services/users';
 import { isAdmin } from 'services/permissions/roles';
 
-declare module 'services/users' {
-  type IProjectFolderModeratorRole = {
-    type: 'project_folder_moderator';
-    project_folder_id: string;
-  };
+interface IProjectFolderModeratorRole {
+  type: 'project_folder_moderator';
+  project_folder_id: string;
```

Comment:
I think this will be an issue. IRole (should actually be called TRole per convention because it's a type) in services/users is not extended this way and I think it should. In the places where we're using IRole, we otherwise won't know of folder moderators, which may cause unexpected bugs.

This role needs to be added via mapping in a module, like we do with e.g. `front/app/modules/commercial/project_folders/services/notifications.ts.`

Reply:
I don't think you can extent a type, right? Only an interface - or am I wrong about that? How would you go about it?
Also this type `IProjectFolderModeratorRole` should only ever be useful inside this module, correct? If it not used elsewhere - how would it cause an issue? Or maybe I'm missing something here..

Reply:
You need to open that link in a new tab or it won't work for some reason.

Reply:
Alright, I think I get it, thanks! Will try to figure it out..

Reply:
Took a bit of Typescript magic (types are not augmentable the same way interfaces are) but it worked in the end. 👍